asreviewcontrib/insights/extract.py

```diff
@@ -0,0 +1,98 @@
+import tempfile
+from pathlib import Path
+
+from asreview import ASReviewData
+from asreview import ASReviewProject
+from asreview import open_state
+
+
+class extract:
+    @staticmethod
+    def extract_and_output(project_file_name, output_path=None, file_format="xlsx"):
+        """
+        Extract data from ASReview state, merge it with the original dataset,
+        and sort it.
+
+        Parameters:
+        - project_file_name: Path to the ASReview project file.
+        - output_path: Optional output path to save the merged dataset.
+        - file_format: Output file format, either 'xlsx' or 'csv'.
```

Comment:
I think the new export API in ASReview largely covers the feature. It is currently disabled but can be enabled easily by removing this code: 

https://github.com/asreview/asreview/blob/8fa7ebce13eb70b64be5a20f68f36edbc82ac9fa/asreview/webapp/api/projects.py#L931-L941